app/main.py

```diff
@@ -1489,35 +1585,43 @@ def update_alert_frame_main(alert_frame_update_new_event, alert_frame_update_int
 @app.callback(
     Output("dashboard_table", "children"),
     Input("interval-component-dashboard-screen", "n_intervals"),
+    [State("user_headers", "data"), State("user_credentials", "data")],
 )
-def update_dashboard_table(n_intervals):
+def update_dashboard_table(n_intervals, user_headers, user_credentials):
     """
     This builds and refreshes the dashboard table to monitor devices status every minute
     Had to fetch devices data again for freshness purposes
     """
-    response = requests.get("https://api.pyronear.org/devices/", headers=api_client.headers)
+    if user_headers is None:
+        raise PreventUpdate
+
+    response = requests.get("https://api.pyronear.org/devices/", headers=user_headers)
     # Check token expiration
     if response.status_code == 401:
-        api_client.refresh_token(cfg.API_LOGIN, cfg.API_PWD)
+        api_client.refresh_token(user_credentials["username"], user_credentials["password"])
         response = requests.get("https://api.pyronear.org/devices/", headers=api_client.headers)
```

Comment:
`/devices`/ route is actually on an admin-scope only. Now that we're using user creds I would recommend to use `/devices/my-devices` that returns user-scoped devices. 

To allow admin reporting (all devices) I would recommend : 

1. Use default `/devices/` route 
2. If not empty --> proceed for admin purposes
3. If empty then use `/devices/my-devices`

Reply:
I completely missed this, thanks!